confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/ConfigNodeProcedureEnv.java

```diff
@@ -40,45 +43,84 @@ public class ConfigNodeProcedureEnv {
+    ConfigNodeProcedureEnv.invalidCacheResult = result;
+  }
+
   public ConfigNodeProcedureEnv(ConfigManager configManager) {
     this.configManager = configManager;
   }
 
-  // TODO: reuse the same ClientPool with other module
-  private static final IClientManager<TEndPoint, SyncDataNodeInternalServiceClient>
-      INTERNAL_SERVICE_CLIENT_MANAGER =
-          new IClientManager.Factory<TEndPoint, SyncDataNodeInternalServiceClient>()
-              .createClientManager(
-                  new DataNodeClientPoolFactory.SyncDataNodeInternalServiceClientPoolFactory());
-
   public ConfigManager getConfigManager() {
     return configManager;
   }
 
-  public InternalService.Client getDataNodeClient(TRegionReplicaSet dataRegionReplicaSet)
-      throws IOException {
-    List<TDataNodeLocation> dataNodeLocations = dataRegionReplicaSet.getDataNodeLocations();
-    int retry = dataNodeLocations.size() - 1;
-    for (TDataNodeLocation dataNodeLocation : dataNodeLocations) {
-      try {
-        return INTERNAL_SERVICE_CLIENT_MANAGER.borrowClient(dataNodeLocation.getInternalEndPoint());
-      } catch (IOException e) {
-        if (retry-- > 0) {
-          LOG.warn(
-              "Connect dataRegion-{} at dataNode-{} failed, trying next replica..",
-              dataRegionReplicaSet.getRegionId(),
-              dataNodeLocation);
-        } else {
-          LOG.warn("Connect dataRegion{} failed", dataRegionReplicaSet.getRegionId());
-          throw e;
-        }
+  /**
+   * Delete config node information, includes (mTree, partitionInfo, regionMap)
+   *
+   * @param deleteSgSchema storage group name
+   * @return tsStatus
+   */
+  public TSStatus deleteConfig(TStorageGroupSchema deleteSgSchema) {
+    DeleteStorageGroupReq deleteStorageGroupReq = new DeleteStorageGroupReq(deleteSgSchema);
+    return configManager.getClusterSchemaManager().deleteStorageGroup(deleteStorageGroupReq);
+  }
+
+  /**
+   * Pre delete a storage group
+   *
+   * @param preDeleteType execute/rollback
+   * @param deleteSgName storage group name
+   */
+  public void preDelete(PreDeleteStorageGroupReq.PreDeleteType preDeleteType, String deleteSgName) {
+    configManager.getPartitionManager().preDeleteStorageGroup(deleteSgName, preDeleteType);
+  }
+
+  /**
+   * @param storageGroupName Storage group name
+   * @return ALL SUCCESS OR NOT
+   * @throws IOException IOE
+   * @throws TException Thrift IOE
+   */
+  public boolean invalidateCache(String storageGroupName) throws IOException, TException {
+    // TODO: Remove it after IT is supported
+    if (skipForTest) {
+      return invalidCacheResult;
+    }
+    List<TDataNodeInfo> allDataNodes = configManager.getNodeManager().getOnlineDataNodes(-1);
+    TInvalidateCacheReq invalidateCacheReq = new TInvalidateCacheReq();
+    invalidateCacheReq.setStorageGroup(true);
+    invalidateCacheReq.setFullPath(storageGroupName);
+    for (TDataNodeInfo dataNodeInfo : allDataNodes) {
```

Comment:
Maybe we can speed up this step by send the request to each DataNode in parallel and wait the result together at the end